src/App/Fossa/YamlDeps.hs

```diff
@@ -37,19 +36,15 @@ import Path
 import Srclib.Converter (depTypeToFetcher)
 import Srclib.Types (AdditionalDepData (..), Locator (..), SourceUnit (..), SourceUnitBuild (..), SourceUnitDependency (SourceUnitDependency), SourceUserDefDep (..))
 
-analyzeFossaDepsYaml :: (Has Diagnostics sig m, Has ReadFS sig m, Has (Lift IO) sig m) => Path Abs Dir -> Maybe ApiOpts -> m [SourceUnit]
+analyzeFossaDepsYaml :: (Has Diagnostics sig m, Has ReadFS sig m, Has (Lift IO) sig m) => Path Abs Dir -> Maybe ApiOpts -> m (Maybe SourceUnit)
 analyzeFossaDepsYaml root maybeApiOpts = do
   maybeDepsFile <- findFossaDepsFile root
   case maybeDepsFile of
-    Nothing -> pure []
-    -- If the file exists and we have no SourceUnit to report, that's a failure
+    Nothing -> pure Nothing
     Just depsFile -> do
       yamldeps <- context "Reading fossa-deps file" $ readContentsYaml depsFile
-      sourceUnit <- context "Converting fossa-deps to partial API payload" $ Just <$> toSourceUnit root yamldeps
-      archiveSrcUnit <- case maybeApiOpts of
-        Nothing -> pure $ archiveNoUploadSourceUnit (vendoredDependencies yamldeps)
-        Just apiOpts -> archiveUploadSourceUnit root apiOpts (vendoredDependencies yamldeps)
-      pure $ catMaybes [sourceUnit, archiveSrcUnit]
+      -- If the file exists and we have no dependencies to report, that's a failure.
```

Comment:
[nit]: This should probably move to where the `when (hasNodeps deps) $ fatalText` call is.